ComfortZoneApp/app/src/main/java/com/example/comfortzone/models/LevelsTracker.java

```diff
@@ -62,6 +65,22 @@ public int getTempAverage() {
         } return MIN_TEMP;
     }
 
+    public int getLowRange() {
+        try {
+            return fetchIfNeeded().getInt(KEY_LOW_RANGE);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        } return MIN_TEMP;
+    }
+
+    public int getHighRange() {
+        try {
+            return fetchIfNeeded().getInt(KEY_HIGH_RANGE);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        } return MAX_TEMP;
+    }
```

Comment:
Since you have this logic in couple places, I wonder if you can write helper method to simplify this:

```
private LevelsTracker fetchIfNeededWrapper() {
       try {
            return fetchIfNeeded();
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return null;
}
```

then whenever you need this, you just do:
```
LevelsTracker tracker = fetchIfNeededWrapper();
return tracker != null ? tracker.getInt(KEY_HIGH_RANGE) : MAX_TEMP;
```

(it'll be much simpler if this is Kotlin):
`return fetchIfNeededWrapper()?.getInt(KEY_HIGH_RANGE) ?: MAX_TEMP;`